google-ads/src/main/java/com/google/ads/googleads/lib/utils/AbstractErrorUtils.java

```diff
@@ -121,15 +122,44 @@
     return result;
   }
 
+  /** Provides a convenience method to get all failed operation indices. */
+  public List<Long> getFailedOperationIndices(GoogleAdsFailureT googleAdsFailureT) {
+    return StreamSupport.stream(getErrorPaths(googleAdsFailureT).spliterator(), false)
+        .filter(ErrorPath::isOperationIndex)
+        .filter(p -> p.getIndex().isPresent())
+        .map(p -> (Long) p.getIndex().get())
+        .distinct()
+        .collect(Collectors.toList());
+  }
+
   /**
    * Unpacks a single {@link GoogleAdsFailureT} from an {@link Any} instance.
    *
-   * @throws InvalidProtocolBufferException if {@link GoogleAdsFailureT} is not able to unpack the
-   *     protocol buffer. This is most likely due to using the wrong version of <code>ErrorUtils
-   *     </code> being used.
+   * @throws DeserializeException if an {@link InvalidProtocolBufferException} is encountered.
    */
-  public GoogleAdsFailureT getGoogleAdsFailure(Any detail) throws InvalidProtocolBufferException {
-    return detail.unpack(getGoogleAdsFailureClass());
+  public GoogleAdsFailureT getGoogleAdsFailure(Any detail) {
+    try {
+      return detail.unpack(getGoogleAdsFailureClass());
+    } catch (InvalidProtocolBufferException e) {
```

Comment:
Is InvalidProtocolBufferException thrown if `detail` is not null but it is not a GoogleAdsFailure?